include/xtensor/xio.hpp

```diff
@@ -203,21 +203,29 @@ namespace xt
 
                 size_type i = 0;
                 size_type elems_on_line = 0;
-                size_type ewp2 = static_cast<size_type>(element_width) + size_type(2);
-                size_type line_lim = static_cast<size_type>(std::floor(line_width / ewp2));
+                size_type const ewp2 = static_cast<size_type>(element_width) + size_type(2);
+                size_type const line_lim = static_cast<size_type>(std::floor(line_width / ewp2));
 
                 out << '{';
                 for (; i != size_type(view.shape()[0] - 1); ++i)
                 {
                     if (edgeitems && size_type(view.shape()[0]) > (edgeitems * 2) && i == edgeitems)
                     {
-                        out << "..., ";
-                        if (view.dimension() > 1)
+                        if (view.dimension() == 1 && line_lim != 0 && elems_on_line >= line_lim)
+                        {
+                            out << " ...,";
+                        }
+                        else if (view.dimension() > 1)
                         {
                             elems_on_line = 0;
-                            out << std::endl
+                            out << "...,"
+                                << std::endl
                                 << indents;
                         }
+                        else
+                        {
+                            out << "..., ";
+                        }
```

Comment:
Non trivial change happening here: removing trailing whitespaces from xio output.

Reply:
I'm good with this. There is just the question if we thinks that anyone depends on the trailing space. Ideas @wolfv @JohanMabille ?

Reply:
I think it could break some tests in downstream projects (like the output tests in xtensor), but nothing hard to fix, so let's merge it.